### PR TITLE
feat: update partner configuration interface for name validation

### PR DESCRIPTION
**Pull request by OnyekaIjeh**

### CODE REVIEW

File: contracts/PartnerConfiguration/PartnerConfiguration.sol [hunk 1]
```diff
@@ -51,6 +64,9 @@ contract PartnerConfiguration is IPartnerConfiguration, Ownable {
        @inheritdoc IPartnerConfiguration
      */
     function setMinLength(uint256 minLength) external override onlyOwner {
+        if (minLength == 0) {
+            revert InvalidMinLength(minLength, "Minimum length cannot be 0");
+        }
```
Comment (sergio-jara) [inline contracts/PartnerConfiguration/PartnerConfiguration.sol:69]: applies here too

File: contracts/PartnerConfiguration/PartnerConfiguration.sol [hunk 2]
```diff
@@ -161,17 +180,32 @@ contract PartnerConfiguration is IPartnerConfiguration, Ownable {
...
         if (duration == 2) return 4 * (10 ** 18);
 
         return (duration + 2) * (10 ** 18);
     }
+
+    /**
+       @inheritdoc IPartnerConfiguration
+     */
+    function isValidName(string calldata name, uint256 duration) external view {
+        if (duration < _minDuration)
```
Comment (sergio-jara) [inline contracts/PartnerConfiguration/PartnerConfiguration.sol:193]: **'<'** or **'<='**?? exclusive or inclusive?? I would say inclusive

File: contracts/PartnerConfiguration/PartnerConfiguration.sol [hunk 3]
```diff
@@ -161,17 +180,32 @@ contract PartnerConfiguration is IPartnerConfiguration, Ownable {
...
+                "Duration less than minimum duration"
+            );
+
+        if ((_maxDuration != 0) && (duration > _maxDuration))
+            revert InvalidDuration(
+                duration,
+                "Duration is more than max duration"
+            );
+
+        if (name.strlen() < _minLength)
```
Comment (sergio-jara) [inline contracts/PartnerConfiguration/PartnerConfiguration.sol:205]: inclusive here too

File: contracts/PartnerConfiguration/PartnerConfiguration.sol [hunk 4]
```diff
@@ -161,17 +180,32 @@ contract PartnerConfiguration is IPartnerConfiguration, Ownable {
...
+       @inheritdoc IPartnerConfiguration
+     */
+    function isValidName(string calldata name, uint256 duration) external view {
+        if (duration < _minDuration)
+            revert InvalidDuration(
+                duration,
+                "Duration less than minimum duration"
+            );
+
+        if ((_maxDuration != 0) && (duration > _maxDuration))
```
Comment (sergio-jara) [inline contracts/PartnerConfiguration/PartnerConfiguration.sol:199]: here

File: contracts/PartnerConfiguration/PartnerConfiguration.sol [hunk 5]
```diff
@@ -161,17 +180,32 @@ contract PartnerConfiguration is IPartnerConfiguration, Ownable {
...
         if (duration == 1) return 2 * (10 ** 18);
         if (duration == 2) return 4 * (10 ** 18);
 
         return (duration + 2) * (10 ** 18);
     }
+
+    /**
+       @inheritdoc IPartnerConfiguration
+     */
+    function isValidName(string calldata name, uint256 duration) external view {
```
Comment (sergio-jara) [inline contracts/PartnerConfiguration/PartnerConfiguration.sol:192]: if it is not going to return a bool, wouldn't it be better to call it "ValidateName"?

**Review by sergio-jara — CHANGES_REQUESTED** [review]

File: contracts/PartnerConfiguration/PartnerConfiguration.sol [hunk 6]
```diff
@@ -28,7 +35,13 @@ contract PartnerConfiguration is IPartnerConfiguration, Ownable {
...
         uint256 minCommitmentAge
     ) {
-        // require(minDuration > 0, "PartnerConfiguration: Invalid min duration");
+        if (minLength == 0) {
+            revert InvalidMinLength(minLength, "Minimum length cannot be 0");
+        }
+
+        if (minDuration == 0) {
+            revert InvalidDuration(minDuration, "Minimum duration cannot be 0");
+        }
```
[the remark this answers is not in the document]
Comment (OnyekaIjeh) [reply contracts/PartnerConfiguration/PartnerConfiguration.sol:44]: Make sense, will add a guard for that.

File: contracts/PartnerConfiguration/PartnerConfiguration.sol (hunk 1 above)
Comment (OnyekaIjeh) [reply contracts/PartnerConfiguration/PartnerConfiguration.sol:69]: Seen.

File: contracts/PartnerConfiguration/PartnerConfiguration.sol (hunk 2 above)
Comment (OnyekaIjeh) [reply contracts/PartnerConfiguration/PartnerConfiguration.sol:193]: Well, I would say `<`, if we set the `minDuration` to be `3` for example, I would assume the `minDuration` can only be 3 and above.

File: contracts/PartnerConfiguration/PartnerConfiguration.sol (hunk 3 above)
Comment (OnyekaIjeh) [reply contracts/PartnerConfiguration/PartnerConfiguration.sol:205]: See above comment

File: contracts/PartnerConfiguration/PartnerConfiguration.sol (hunk 4 above)
Comment (OnyekaIjeh) [reply contracts/PartnerConfiguration/PartnerConfiguration.sol:199]: See above comment

File: contracts/PartnerConfiguration/PartnerConfiguration.sol (hunk 5 above)
Comment (OnyekaIjeh) [reply contracts/PartnerConfiguration/PartnerConfiguration.sol:192]: I agree, I was going to raise it but wanted to follow what we agreed initially. Will change the name.
Thank you.

File: contracts/PartnerConfiguration/PartnerConfiguration.sol (hunk 2 above)
Comment (sergio-jara) [reply contracts/PartnerConfiguration/PartnerConfiguration.sol:224]: yes. you are right.

**Review by sergio-jara — APPROVED** [review]

**Review by sergio-jara — APPROVED** [review]

**Pull request merged**